### DISCUSSION TITLE
What do the "Automatic song picker" and "Original algorithm for recommendations" settings do?

### DISCUSSION THREAD

**Discussion by natema** [gokadzev/Musify#412] [Q&A] [answered]
![2024-08-15 18 35 30](https://github.com/user-attachments/assets/5688ab50-8c6a-410e-a4fe-00c97aea9f1f)
I didn't manage to find an explanation for what these settings do exactly (even though I have a guess for the first, based on the name). 
How do they change the app behavior exactly?

**Comment by gokadzev** [comment, marked as answer]
> @natema I didn't manage to find an explanation for what these settings do exactly (even though I have a guess for the first, based on the name). How do they change the app behavior exactly?

- The automatic song picker selects a random but globally popular song if there is only one song in the queue (technically, it functions as a non-stop queue with randomized songs).
- The original algorithm for recommendations is YouTube's algorithm, which is based on your recent played several songs. If it is turned off, you will receive app custom recommendations that are based on your listening history and popular songs (just not higher level implementation).

**Reply by natema** [reply]
I guess that YouTube's recommendations are better when one is logged in, and one would also have the history of its played tracks and continue to calibrate their algorithm. 
Would it be a reasonable feature request to add the possibility be signed in with a Google Account while using the app?
